orttraining/orttraining/core/optimizer/graph_transformer_utils.cc

```diff
@@ -99,6 +100,7 @@ std::vector<std::unique_ptr<GraphTransformer>> GeneratePreTrainingTransformers(
       transformers.emplace_back(std::make_unique<SimplifiedLayerNormFusion>(compatible_eps));
       transformers.emplace_back(std::make_unique<FastGeluFusion>(compatible_eps));
       transformers.emplace_back(std::make_unique<SoftmaxCrossEntropyLossInternalFusion>(compatible_eps));
+      transformers.emplace_back(std::make_unique<GatherToSplitFusion>(compatible_eps));
```

Comment:
When are the transformers under "GeneratePreTrainingTransformers" applied?

Reply:
looks like they are applied as level 1 transformers. am I right?

Reply:
They are applied before gradient builder for training case. Normally if you want it take effect before gradient builder so that gradient builder can use it you need to add it here. E.g., we fuse Gather nodes to Split, then gradient builder will generate Split's gradient, instead of Gather's gradients. You may also need to add it to the file in onnxruntime folder so that it can also help the inference scenario.

Yes, it's at level 1.